RavenDB-14758 NRE in GetMinLastEtag if backup is not created

ayende opened the pull request.

grisha-kotler reviewed (approved):
LGTM

ppekrol commented:
Tests are failing because of this change

grisha-kotler commented on src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:
```diff
@@ -688,6 +688,11 @@ private long GetMinLastEtag()
+                    if (status == null)
+                    {
+                        // if there is no status for this, we don't need to take into account tombstones
+                        continue;
```
```suggestion
                        min = 0; // cannot delete the tombstones until we've done a full backup
                        continue;
```

grisha-kotler commented on src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:
```diff
@@ -688,6 +688,11 @@ private long GetMinLastEtag()
                         continue; // if the backup is always full, we don't need to take into account the tombstones, since we never back them up.
```
This change should also be here

ayende committed:
RavenDB-14758 NRE in GetMinLastEtag if backup is not created

ppekrol commented:
We are still having issue with 1 test after this change

ayende committed:
RavenDB-14758 We should return etag 0 only for backups that never run, not for full backups
Responsibility for last etag on tombstones for full backups is on the other members (etl, external replication, etc)
